File: Interview_Prep_App/native_java/javatemplate/src/main/java/com/tc2r/javatemplate/repositories/models/MultipleChoiceQuestion.java
package com.tc2r.javatemplate.repositories.models;

import java.util.List;

/**
 * Created by Tc2r on 5/19/2017.
 * <p>
 * Description:
 */

public class MultipleChoiceQuestion {
    private final String question;
    private final List<Answer> answerList;

    public String getQuestion() {
        return question;
    }

    public List<Answer> getAnswers() {
        return answerList;
    }


    public static class Builder {
        // Required Parameters
        private String question;
        private List<Answer> answerList;


        public Builder() {}

        public Builder possibleAnswers(List<Answer> listOfAnswers) {
            this.answerList = listOfAnswers;
            return this;
        }

        public Builder quizQuestion(String question) {
            this.question = question;
            return this;
        }

        public MultipleChoiceQuestion build() {
            return new MultipleChoiceQuestion(this);
        }
    }

    private MultipleChoiceQuestion(Builder builder) {
        question = builder.question;
        answerList = builder.answerList;
    }
}

File: Interview_Prep_App/native_kotlin/base/src/main/java/com/nudennie/interviewprepapp/repositories/models/MultipleChoiceQuestion.kt
package com.nudennie.interviewprepapp.repositories.models

import com.nudennie.interviewprepapp.repositories.models.response.Answer

/**
 * Created by Tc2r on 5/19/2017.
 *
 *
 * Description:
 */
class MultipleChoiceQuestion private constructor(builder: Builder) {
    val question: String
    val answers: List<Answer>

    class Builder {
        // Required Parameters
        internal var question: String = ""
        internal var answerList: List<Answer> = emptyList()

        // Function to assign all the answer choices for a question
        fun possibleAnswers(listOfAnswers: List<Answer>): Builder {
            answerList = listOfAnswers
            return this
        }

        // Function to assign question
        fun quizQuestion(question: String): Builder {
            this.question = question
            return this
        }

        fun build(): MultipleChoiceQuestion {
            return MultipleChoiceQuestion(this)
        }
    }

    // Initialising the question and the answers
    init {
        question = builder.question
        answers = builder.answerList
    }
}
